### PR TITLE
Bug fixes mainly for network status

### PR DESCRIPTION
**Pull request by wiverson**
## What kind of change does this PR introduce?

Bug fixes, mainly for the network status. Minor cleanup for refresh token thread.

### CODE REVIEW

**Comment by wiverson** [comment]
@acupofjose did you want to sanity check these before I merge? Kind of like having a second pair of eyes but just lmk. :)

File: Gotrue/NetworkStatus.cs [hunk 1]
```diff
@@ -13,58 +15,114 @@ namespace Supabase.Gotrue
...
 
+		/// <summary>
+		/// The <see cref="Client"/> that this network status system is attached to.
+		/// </summary>
+		public IGotrueClient<User, Session>? Client { get; set; }
+
 		/// <summary>
 		/// Pings the URL in the <see cref="Client.Options"/> to check if the network is online.
+		/// 
+		/// https://wutdruyqartmqvkdaxnf.supabase.co/auth/v1/settings
```
Comment (acupofjose) [inline Gotrue/NetworkStatus.cs:64]: Probably want to remove your instance here!

**Review by acupofjose — APPROVED** [review]
Just needs that comment adjusted - otherwise looks sane to me!

File: Gotrue/NetworkStatus.cs (hunk 1 above)
Comment (wiverson) [reply Gotrue/NetworkStatus.cs]: lol thanks!

**Pull request merged**